core/src/main/java/greencity/webcontroller/ManagementUserController.java

```diff
@@ -76,13 +78,27 @@ public String updateUser(@Valid @ModelAttribute("user") UserManagementDto userDt
      * Method for finding {@link User} by id.
      *
      * @param id of the searched {@link User}.
-     * @return dto {@link UserForListDto} of the {@link User}.
+     * @return dto {@link UserManagementDto} of the {@link User}.
      * @author Vasyl Zhovnir
      */
     @GetMapping("/findById")
     @ResponseBody
-    public UserForListDto findById(Long id) {
+    public UserManagementDto findById(Long id) {
         User byId = userService.findById(id);
-        return modelMapper.map(byId, UserForListDto.class);
+        return modelMapper.map(byId, UserManagementDto.class);
+    }
+
+    /**
+     * Method for setting {@link User}'s status to DEACTIVATED,
+     * so the user will not be able to log in into the system.
+     *
+     * @param id of the searched {@link User}.
+     * @author Vasyl Zhovnir
+     */
+    @PostMapping
+    public ResponseEntity<UserDeactivateDto> deactivateUser(Long id) {
+        User byId = userService.findById(id);
+        UserDeactivateDto dto = modelMapper.map(byId, UserDeactivateDto.class);
```

Comment:
You are: 

- extracting the user from the database by _id_
- converting to dto and pass it to `userService.deactivateUser(dto)`
- finally ... in method `deactivateUser` get only the _id_ from the dto to find user once again

I think you can pass the _id_ directly to service method `deactivateUser`

Reply:
Thank you, done.